common/src/main/java/com/wynntils/models/items/encoding/impl/block/ShinyDataTransformer.java

```diff
@@ -7,14 +7,39 @@
 import com.wynntils.models.items.encoding.data.ShinyData;
 import com.wynntils.models.items.encoding.type.DataTransformer;
 import com.wynntils.models.items.encoding.type.ItemTransformingVersion;
+import com.wynntils.utils.UnsignedByteUtils;
 import com.wynntils.utils.type.ErrorOr;
 import com.wynntils.utils.type.UnsignedByte;
 
-public class ShinyDataTransformer implements DataTransformer<ShinyData> {
+public class ShinyDataTransformer extends DataTransformer<ShinyData> {
+    public static final byte ID = 6;
+
     @Override
     public ErrorOr<UnsignedByte[]> encodeData(ItemTransformingVersion version, ShinyData data) {
         return switch (version) {
-            case VERSION_1 -> ErrorOr.of(new UnsignedByte[0]);
+            case VERSION_1 -> ErrorOr.of(encodeShinyData(data));
         };
     }
+
+    @Override
+    protected boolean shouldEncodeData(ItemTransformingVersion version, ShinyData data) {
+        return data.shinyStat() != null && data.shinyStat().statType() != null;
+    }
+
+    @Override
+    protected byte getId() {
+        return ID;
+    }
+
+    private static UnsignedByte[] encodeShinyData(ShinyData data) {
+        UnsignedByte[] shinyStatValueBytes =
+                UnsignedByteUtils.encodeVariableSizedInteger(data.shinyStat().value());
+
+        UnsignedByte[] bytes = new UnsignedByte[shinyStatValueBytes.length + 2];
```

Comment:
All the other non-trivial encodes had nice comments describing the format. I'd like to see that here to, it was very helpful.